### PR TITLE
Feat/invidious

### LINKED ISSUE #627: invidious api
What about using invidious api =>  https://github.com/omarroth/invidious
and use youtube plugin as a fallback when something fail.

https://github.com/FreeTubeApp/FreeTube use it and it works just fine.

This could be configurable in the settings:
 - invidous instance url
 - enable/disable

### PR DESCRIPTION
**Pull request by charjac**
=== App ===
 - move  all env variable to root .env file
 - add invidious url in youtube settings section with default to https://invidio.us
 - add invdious plugin
 - if invidious fail, fallback with youtube plugin
 - use invidious by default

fix #627

### CODE REVIEW

File: packages/app/app/containers/HttpApiUrl/index.js [hunk 1]
```diff
@@ -15,7 +15,6 @@ const HttpApiUrl = ({ settings }) => {
   return (
     <a
       className={styles.http_api_url}
-      href='javascript:void(0)'
```
Comment (nukeop) [inline packages/app/app/containers/HttpApiUrl/index.js]: I think this href was needed for some cosmetic reason, did you notice anything different when removing it?

File: packages/app/app/containers/SoundContainer/index.js [hunk 2]
```diff
@@ -130,6 +137,14 @@ class SoundContainer extends React.Component {
     });
   }
 
+  handleError() {
```
Comment (nukeop) [inline packages/app/app/containers/SoundContainer/index.js]: hacky, maybe it'd be better to store a fallback/default plugin in redux (like we store the selected one) and use it when errors occur?

File: packages/app/app/locales/en.json [hunk 3]
```diff
@@ -191,6 +191,7 @@
     "user": "User:",
     "youtube": "Youtube",
     "yt-api-key": "Youtube API Key",
+    "invidious-url": "Invidious instance url",
```
Comment (nukeop) [inline packages/app/app/locales/en.json]: This is ok for now but I'm making a mental note to introduce per-plugin settings at some point and configure stuff like this in the plugins section

File: packages/app/app/plugins/stream/InvidiousPlugin.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import logger from 'electron-timber';
+
+import StreamProviderPlugin from '../streamProvider';
+import * as Invidious from '../../rest/Invidious';
+import YoutubePlugin from './YoutubePlugin';
+
+class InvidiousPlugin extends StreamProviderPlugin {
```
Comment (nukeop) [inline packages/app/app/plugins/stream/InvidiousPlugin.js]: I think I'll define these things properly in Typescript soon

File: packages/app/app/locales/en.json (hunk 3 above)
Comment (charjac) [reply packages/app/app/locales/en.json]: sound great

File: packages/app/app/containers/SoundContainer/index.js (hunk 2 above)
Comment (charjac) [reply packages/app/app/containers/SoundContainer/index.js]: sure, one more select in the plugins view ? or something hard coded ?

File: packages/app/app/containers/HttpApiUrl/index.js (hunk 1 above)
Comment (charjac) [reply packages/app/app/containers/HttpApiUrl/index.js]: i replace it with css style 'cursor: pointer;'. it remove the annoying warning in console

File: packages/app/app/plugins/stream/InvidiousPlugin.js (hunk 4 above)
Comment (charjac) [reply packages/app/app/plugins/stream/InvidiousPlugin.js]: I was also thinking about moving reducer and action constants in @nuclear/core, without any electron reference (mobile app in mind).
What do you think about it ?

File: packages/app/app/containers/HttpApiUrl/index.js (hunk 1 above)
Comment (charjac) [reply packages/app/app/containers/HttpApiUrl/index.js]: I replace it with css style
```css
cursor: pointer;
```
it remove the annoying warning

File: packages/app/app/plugins/stream/InvidiousPlugin.js (hunk 4 above)
Comment (nukeop) [reply packages/app/app/plugins/stream/InvidiousPlugin.js]: We can still use reducers on mobile (I think) but decoupling electron dependencies is a step in a good direction

File: packages/app/app/containers/SoundContainer/index.js (hunk 2 above)
Comment (nukeop) [reply packages/app/app/containers/SoundContainer/index.js]: Can be hardcoded somewhere, I just don't like depending on comparing string names

**Comment by charjac** [comment]
I add a few things :) : 
 - queue items have two more optionnal property:
   - failed: bool
   - selectedStream: string
 - new property fallback on StreamProviders plugins
 - compute the stream depending on these 3 params and global selected streamProviders
 - fix the queue Popup stream selector (without breaking the alternate stream this time)

**Comment by nukeop** [comment]
What happens if a queue item is failed?

**Comment by charjac** [comment]
it switch to the fallback define in the used StreamProviders plugin

**Comment by nukeop** [comment]
Ok, this is a very good solution

**Comment by nukeop** [comment]
Do you know why the tests are failing?

**Comment by charjac** [comment]
no will check locally

**Comment by charjac** [comment]
same error locally, i will fix it

**Comment by charjac** [comment]
queureducer now import an actions from plugins, and plugins actions require something from electron package. Mocha didnt like it. electron-mocha seems to be ok.

**Comment by charjac** [comment]
ok its fucked on ci .... will find something

**Comment by nukeop** [comment]
Maybe remove the electron dependency and inject at runtime.

**Comment by charjac** [comment]
does plugins still need to access the store from window ?

**Comment by nukeop** [comment]
It's for the arbitrary plugins loaded by users. They get the app instance, the store, and the React api. But all this can be passed to the action as arguments and kept out of the action definitions.

**Comment by charjac** [comment]
finally i just mock electron related modules and test passes

**Pull request merged**